packages/algob/src/internal/deployer.ts

```diff
@@ -742,33 +744,62 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
    * https://developer.algorand.org/docs/features/asc1/stateless/sdks/#account-delegation-sdk-usage
    * @param name: Logic Signature filename (must be present in assets folder)
    * @param signer: Signer Account which will sign the smart contract
+   * @param lsigName name of smart signature (checkpoint info will be stored against this name)
    * @param scTmplParams: scTmplParams: Smart contract template parameters
    *     (used only when compiling PyTEAL to TEAL)
-   * @param lsigName name of smart signature (if passed, checkpoint info will be stored against this name)
    */
   async mkDelegatedLsig (
     fileName: string,
+    lsigName: string,
     signer: rtypes.Account,
-    scTmplParams?: SCParams,
-    lsigName?: string
+    scTmplParams?: SCParams
   ): Promise<LsigInfo> {
     return await this._mkLsig(fileName, scTmplParams, signer, lsigName);
   }
 
   /**
    * Stores logic signature info in checkpoint for contract mode
    * @param fileName ASC file name
+   * @param lsigName name of lsig (checkpoint info will be stored against this name)
    * @param scTmplParams: Smart contract template parameters (used only when compiling PyTEAL to TEAL)
-   * @param lsigName name of lsig (if passed, checkpoint info will be stored against this name)
    */
   async mkContractLsig (
     fileName: string,
-    scTmplParams?: SCParams,
-    lsigName?: string
+    lsigName: string,
+    scTmplParams?: SCParams
   ): Promise<LsigInfo> {
     return await this._mkLsig(fileName, scTmplParams, undefined, lsigName);
   }
 
+  /**
+   * Create and sign (using signer's sk) a logic signature for "delegated approval". Then save signed lsig
+   * info to checkpoints (in /artifacts)
+   * https://developer.algorand.org/docs/features/asc1/stateless/sdks/#account-delegation-sdk-usage
+   * @param name: Stateless Smart Contract filename (must be present in assets folder)
+   * @param signer: Signer Account which will sign the smart contract
+   * @param scTmplParams: scTmplParams: Smart contract template parameters
+   *     (used only when compiling PyTEAL to TEAL)
+   */
+  async mkDelegatedLsigByFile (
+    fileName: string,
+    signer: rtypes.Account,
+    scTmplParams?: SCParams
+  ): Promise<LsigInfo> {
+    return await this._mkLsig(fileName, scTmplParams, signer);
+  }
+
+  /**
+   * Stores logic signature info in checkpoint for contract mode
+   * @param fileName ASC file name
+   * @param scTmplParams: Smart contract template parameters (used only when compiling PyTEAL to TEAL)
+   */
+  async mkContractLsigByFile (
+    fileName: string,
+    scTmplParams?: SCParams
+  ): Promise<LsigInfo> {
+    return await this._mkLsig(fileName, scTmplParams, undefined);
+  }
```

Comment:
both `mkContractLsig` and `mkContractLsigByFile` create an  lsig by flename. Maye we can remove the alias and improve the documentation that the name can be optinal - in that case we shouldn't change the order of arguments.

Reply:
This is how we were doing it before. but in the discord thread we discussed we should make "by name" behaviour as the default. So we created alias functions for files (`eg. mkContractLsigByFile`), and update the default function to use `name` as a required argument. Let's follow what we discussed.